hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -435,28 +486,60 @@ public void refreshSources(String peerId) throws IOException {
    * @param src source to clear
    */
   void removeRecoveredSource(ReplicationSourceInterface src) {
-    LOG.info("Done with the recovered queue " + src.getQueueId());
+    LOG.info("Done with recovered replication queue {}", src.getStoreQueueId());
     this.oldsources.remove(src);
     // Delete queue from storage and memory
-    deleteQueue(src.getQueueId());
-    this.walsByIdRecoveredQueues.remove(src.getQueueId());
+    deleteQueue(src);
+    this.walsByIdRecoveredQueues.remove(src.getStoreQueueId());
   }
 
   /**
    * Clear the metrics and related replication queue of the specified old source
-   * @param src source to clear
    */
-  void removeSource(ReplicationSourceInterface src) {
-    LOG.info("Done with the queue " + src.getQueueId());
+  public void removeSource(ReplicationSourceInterface src) {
+    LOG.info("Done with replication queue {}", src.getStoreQueueId());
     this.sources.remove(src.getPeerId());
     // Delete queue from storage and memory
-    deleteQueue(src.getQueueId());
-    this.walsById.remove(src.getQueueId());
+    deleteQueue(src);
+    this.walsById.remove(src.getStoreQueueId());
   }
 
   /**
-   * Delete a complete queue of wals associated with a replication source
-   * @param queueId the id of replication queue to delete
+   * Add hbase:meta replication source. Called on open of hbase:meta.
+   * @see #removeHBaseMetaSource()
+   */
+  public ReplicationSourceInterface addHBaseMetaSource() throws IOException {
```

Comment:
It is because this method? Then let's add a synchornized in this method and do not change the logic of the addSource method?

Reply:
Not because of this method, no. Was a general concern adding sources.

Let me restore the old code. This is not a critical change.